### ISSUE TITLE
Inconsistent error message from `is_process_alive`

### ISSUE THREAD

**Issue by nathanl** [erlang/otp#8559]
**Describe the bug**

`is_process_alive/1` can raise the error "not a pid" for a value for which `is_pid/1` returns `true`.
This happens when the pid is not local.

**To Reproduce**

1. Start two nodes:

```
erl -sname one@localhost -setcookie hello
```
and
```
erl -sname two@localhost -setcookie hello
```

2. Spawn a process on the other node to get a valid remote pid.

```erlang
(one@localhost)1> Pid = spawn('two@localhost', fun() -> 'hi' end).
<10531.109.0>
```

3. Attempt to work with that pid.

```erlang
(one@localhost)2> is_pid(Pid).
true
(one@localhost)3> is_process_alive(Pid).
** exception error: bad argument
     in function  is_process_alive/1
        called as is_process_alive(<10531.109.0>)
        *** argument 1: not a pid
```

**Expected behavior**

The error message should say "not a local pid", or "is a remote pid", or something else that explains why it's not possible to check whether it's alive. "not a pid" is untrue and is inconsistent with the return value from `is_pid/1`.

**Affected versions**
Tested with OTP 27.0

**Comment by garazdawi** [comment]
The code that prints that value is located [here](https://github.com/erlang/otp/blob/OTP-27.0/lib/kernel/src/erl_erts_errors.erl#L473-L474). A PR improving it would be most welcome!

**Comment by garazdawi** [comment]
Fix merged for release in 27.1
